### PR TITLE
fix(frontend): token balance could be null or undefined

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

There is a difference between an `undefined` balance and a `null` balance. The balance is `undefined` if the balance store is not initiated or the specific balance loader for the token is not initiated. But, if the balance loader was initiated at some point, it will either contain data or be `null`, but not `undefined`.

We distinguish the two cases to make it easier to understand when a Token could still need to wait for the balance loader to exists, or if the balance loader is simply not working (disconnected/stopped/error/etc).

### CODE REVIEW

File: src/frontend/src/lib/types/token.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export type OptionTokenStandard = TokenStandard | undefined | null;
 export type TokenToPin = Pick<Token, 'id'> & { network: Pick<Token['network'], 'id'> };
 
 interface TokenFinancialData {
-	balance?: BigNumber;
+	balance?: BigNumber | null;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/types/token.ts:44]: Just an idea that crossed my mind and not for this PR: maybe we can create a TypeScript util for those nullable types?

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

Nitpick: Regarding title of the PR, isn't this more a fix than a feature?

File: src/frontend/src/lib/types/token.ts (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/types/token.ts:44]: similar to `mapAddress` you mean? generalize it?

```
export const mapAddress = <T extends Address>(
	$addressStore: StorageAddressData<T>
): OptionAddress<Address> => ($addressStore === null ? null : $addressStore?.data);
```

Comment (peterpeterparker) [reply src/frontend/src/lib/types/token.ts:44]: No I meant a type, more like `RequiredExcept` or `ResultSuccessReduced`.

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/types/token.ts:44]: Started working on it with:

https://github.com/dfinity/oisy-wallet/pull/2326

**Pull request merged**